packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -118,46 +118,58 @@ contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled, OVM_Testable {
+     * @param _minimumBridgingDelay the new minimum delay.
      */
-    function setMinimumBridgingDelay(uint64 _MinimumBridgingDelay) public onlyFromCrossDomainAccount(bridgeRouter) {
-        _setMinimumBridgingDelay(_MinimumBridgingDelay);
+    function setMinimumBridgingDelay(uint64 _minimumBridgingDelay) public onlyFromCrossDomainAccount(bridgeAdmin) {
+        _setMinimumBridgingDelay(_minimumBridgingDelay);
     }
 
     /**
      * @notice Enables L1 owner to whitelist a L1 Token <-> L2 Token pair for bridging.
-     * @dev Only callable by the existing bridgeRouter via the optimism cross domain messenger.
+     * @dev Only callable by the existing bridgeAdmin via the optimism cross domain messenger.
      * @param l1Token Address of the canonical L1 token. This is the token users will receive on Ethereum.
      * @param l2Token Address of the L2 token representation. This is the token users would deposit on optimism.
+     * @param l1BridgePool Address of the L1 withdrawal pool linked to this L2+L1 token.
      */
-    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(bridgeRouter) {
+    function whitelistToken(
+        address l1Token,
+        address l2Token,
+        address l1BridgePool
+    ) public onlyFromCrossDomainAccount(bridgeAdmin) {
         whitelistedTokens[l2Token] = L2TokenRelationships({
             l1Token: l1Token,
-            lastBridgeTime: uint64(getCurrentTime())
+            l1BridgePool: l1BridgePool,
+            lastBridgeTime: uint64(getCurrentTime()),
+            depositsEnabled: true
         });
 
-        emit WhitelistToken(l1Token, l2Token, uint64(getCurrentTime()));
+        emit WhitelistToken(l1Token, l2Token, uint64(getCurrentTime()), l1BridgePool);
     }
 
+    // TODO: Also provide method to blanket enable/disable entire contract? We might want both that functionality
+    // along with the ability to enable/disable individual L2 tokens.
+
     /**
-     * @notice L1 owner can enable/disable deposits over all whitelisted tokens.
-     * @dev Only callable by the existing bridgeRouter via the optimism cross domain messenger.
+     * @notice L1 owner can enable/disable deposits for a whitelisted tokens.
+     * @dev Only callable by the existing bridgeAdmin via the optimism cross domain messenger.
+     * @param _l2Token address of L2 token to enable/disable deposits for.
      * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
      */
-    function setEnableDeposits(bool _depositsEnabled) public onlyFromCrossDomainAccount(bridgeRouter) {
-        depositsEnabled = _depositsEnabled;
-        emit DepositsEnabled(_depositsEnabled);
+    function setEnableDeposits(address _l2Token, bool _depositsEnabled) public onlyFromCrossDomainAccount(bridgeAdmin) {
+        L2TokenRelationships storage _token = whitelistedTokens[_l2Token];
```

Comment:
do you have to declare the `storage` object like this? could you not just do:
```
whitelistedTokens[_l2Token].depositsEnabled = _depositsEnabled;
```

Reply:
Good call, done